sdcm/cluster_gce.py

```diff
@@ -311,14 +311,17 @@ def set_tags_as_labels(_instance):
         # Name must start with a lowercase letter followed by up to 63
         # lowercase letters, numbers, or hyphens, and cannot end with a hyphen
         assert len(name) <= 63, "Max length of instance name is 63"
-        startup_script = self.test_config.get_startup_script()
+        startup_script = self.test_config.get_startup_script(
```

Comment:
get_startup_script is being called from multiple places
Did you run the ARM tests? They are running in AWS, seems it's not currently handled

Also when I think about it, we might need to do the figuring out inside the startup script, casue what would happen when our AMI would move to that Ubuntu version (also any older distro might get ssh update, regardless)

Reply:
Agreed about moving the check inside the script. It's condition is now for ssh version to be greater than 8.8.

As for the ARM tests: the test for the previous Ubuntu release (20.04) work without any issues. Ubuntu 22.04 however bumps into the same issue as the regular GCE test had - ssh-rsa key format not being supported. It looks like the user-data script that resolves the issue on GCE is not being executed at all on AWS. Therefore we can't modify `sshd_config` and the job fails while waiting for ssh to be up.

Reply:
@roydahan @fruch I created a new issue for our config script: https://github.com/scylladb/scylla-cluster-tests/issues/4818 This is also related to a `scylla-machine-image` issue Łukasz brought up: https://github.com/scylladb/scylla-machine-image/issues/357 In short, since we're not taking advantage of `cloud-init` and delegating config operations to `scylla-cloud-image`, we're opening ourselves to issues like the one with `openssh`. This particular one can be solved by changing our ssh keys, but in general it would be great to make use of `cloud-init`.

Reply:
the bug that SCT breaks the support of cloud-init, it's directly related to scylla-machine-image. it would be nice that it would be the same, but we need to address it SCT regardless of a change in machine-image.